tests/python_package_test/test_dask.py

```diff
@@ -180,7 +283,49 @@ def test_regressor_local_predict(client, listen_port):
 
     # Predictions and scores should be the same
     assert_eq(p1, p2)
-    assert_eq(s1, s2)
+    assert np.isclose(s1, s2, rtol=1e-4)
+
+
+@pytest.mark.parametrize('output', ['array', 'dataframe'])
+def test_ranker(output, client, listen_port):
+    X, y, w, g, dX, dy, dw, dg = _create_ranking_data(output=output)
```

Comment:
could you add one more test like this, but where the groups have different numbers of records? Like 100 records with `group = [5, 15, 25, 5, 5, 5, 10, 10, 10, 10]` or something? There's such a tight relationship here between the group setup and the way the Dask collection is partitioned, I think a test like that could catch some types of issues not caught by the current tests.

Reply:
Yeah definitely good idea. A lot of the state of my test_dask.py reflects me dumbing down the tests trying to get them to pass the "Binding port" bug.

Reply:
Addressed in d6e5209

Reply:
thanks!